cli/src/pcluster/models/cluster.py

```diff
@@ -624,15 +620,30 @@ def _register_validators(self):
     @property
     def architecture(self):
         """Compute cluster's architecture based on its head node instance type."""
-        supported_architectures = get_supported_architectures_for_instance_type(self.instance_type)
-        if not supported_architectures:
-            error(f"Unable to get architectures supported by instance type {self.instance_type}")
-        # If the instance type supports multiple architectures, choose the first one.
-        # TODO: this is currently not an issue because none of the instance types we support more than one of the
-        #       architectures we support. If this were ever to change (e.g., we start supporting i386) then we would
-        #       probably need to choose based on the subset of the architectures supported by both the head node and
-        #       compute instance types.
-        return supported_architectures[0]
+        instance_type_info = AWSApi.instance().ec2.get_instance_type_info(self.instance_type)
+        return instance_type_info.supported_architecture()
+
+    @property
+    def vcpus(self):
+        """Get the number of vcpus for the instance according to disable_hyperthreading and instance features."""
+        instance_type_info = AWSApi.instance().ec2.get_instance_type_info(self.instance_type)
+        default_threads_per_core = instance_type_info.default_threads_per_core()
+        return (
+            instance_type_info.vcpus_count()
+            if self.simultaneous_multithreading
+            else (instance_type_info.vcpus_count() // default_threads_per_core)
+        )
+
+    @property
+    def pass_cpu_options_in_launch_template(self):
+        """Check whether CPU Options must be passed in launch template for head node."""
+        instance_type_info = AWSApi.instance().ec2.get_instance_type_info(self.instance_type)
+        return not self.simultaneous_multithreading and instance_type_info.is_cpu_options_supported_in_lt()
+
+    @property
+    def instance_type_info(self):
+        """Return head node instance type information as returned from aws ec2 describe-instamce-types."""
```

Comment:
You defined it but you're not using it for pass_cpu_options_in_launch_template, nor vcpus, nor architecture.

Then: minor typo in the comment.